quickwit/quickwit-doc-mapper/src/tokenizers.rs

```diff
@@ -17,34 +17,279 @@
+
+        while let Some((offset_from, c)) = iter.next() {
+            match char_grouping(c) {
+                Grouping::Keep => {
+                    let offset_to = if let Some((next_index, next_char)) =
+                        iter.find(|&(_, c)| char_grouping(c) != Grouping::Keep)
+                    {
+                        self.last_char = Some((next_index, next_char));
+                        next_index
+                    } else {
+                        self.text.len()
+                    };
+
+                    self.token.offset_from = offset_from;
+                    self.token.offset_to = offset_to;
+                    self.token.text.push_str(&self.text[offset_from..offset_to]);
+                    return true;
+                }
+                Grouping::SplitKeep => {
+                    let len = c.len_utf8();
+                    self.token.offset_from = offset_from;
+                    self.token.offset_to = offset_from + len;
+                    self.token
+                        .text
+                        .push_str(&self.text[offset_from..(offset_from + len)]);
+                    return true;
+                }
+                Grouping::SplitIgnore => (),
+            }
+        }
+        false
+    }
+
+    fn token(&self) -> &Token {
+        &self.token
+    }
+
+    fn token_mut(&mut self) -> &mut Token {
+        &mut self.token
+    }
+}
+
 /// Quickwits default tokenizer
 pub static QUICKWIT_TOKENIZER_MANAGER: Lazy<TokenizerManager> =
     Lazy::new(get_quickwit_tokenizer_manager);
 
-#[test]
-fn raw_tokenizer_test() {
-    let my_haiku = r#"
+#[cfg(test)]
+mod tests {
+    use tantivy::tokenizer::Token;
+
+    use super::get_quickwit_tokenizer_manager;
+
+    #[test]
+    fn test_raw_tokenizer() {
+        let my_haiku = r#"
         white sandy beach
         a strong wind is coming 
         sand in my face
         "#;
-    let my_long_text = "a text, that is just too long, no one will type it, no one will like it, \
-                        no one shall find it. I just need some more chars, now you may not pass.";
-
-    let tokenizer = get_quickwit_tokenizer_manager().get("raw").unwrap();
-    let mut haiku_stream = tokenizer.token_stream(my_haiku);
-    assert!(haiku_stream.advance());
-    assert!(!haiku_stream.advance());
-    assert!(!tokenizer.token_stream(my_long_text).advance());
+        let my_long_text = "a text, that is just too long, no one will type it, no one will like \
+                            it, no one shall find it. I just need some more chars, now you may \
+                            not pass.";
+
+        let tokenizer = get_quickwit_tokenizer_manager().get("raw").unwrap();
+        let mut haiku_stream = tokenizer.token_stream(my_haiku);
+        assert!(haiku_stream.advance());
+        assert!(!haiku_stream.advance());
+        assert!(!tokenizer.token_stream(my_long_text).advance());
+    }
+
+    #[test]
+    fn test_chinese_tokenizer() {
```

Comment:
Can you add a proptest that check that we get the same results than the regular tokenizer if we only use regular ascii cars?